### PR TITLE
Integrate with Github CI

### PR DESCRIPTION
**Pull request by almet**
- [x] Create Github Actions to run the tests
- [x] Make github actions use postgres
- [x] Make Playwright work
- [x] Support for python 3.12
- [x] Format the files 
- [x] Make the linter happy

### CODE REVIEW

**Comment by almet** [comment]
I don't understand why the CI is not using the proper database configuration.

image

shows that it uses the "postgres" database, whereas I'm defining this otherwise in `tests/settings.py`:

```python

if os.environ.get("GITHUB_ACTIONS", False) == "true":
    DATABASES = {
        "default": {
            "ENGINE": "django.contrib.gis.db.backends.postgis",
            "NAME": "umap",
            "USER": "umap",
            "PORT": 5432,
            "PASSWORD": "umap",
        }
    }
```

**Comment by almet** [comment]
The error is now, trying to fix it by specifying the `HOST`.
```
psycopg2.OperationalError: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
```

**Comment by almet** [comment]
Now upgrading the postgres version to 12, as its the minimum required version for Django.

**Comment by almet** [comment]
Now installing playwright :-)

**Comment by almet** [comment]
Playwright is failing with stuff like:

```
______________________ test_map_load_with_owner[chromium] ______________________

anonymap = <Map: test map>
live_server = <LiveServer listening at http://localhost:50485>
owner_session = <Page url='http://localhost:50485/en/map/test-map_141'>

    def test_map_load_with_owner(anonymap, live_server, owner_session):
        owner_session.goto(f"{live_server.url}{anonymap.get_absolute_url()}")
        map_el = owner_session.locator("#map")
        expect(map_el).to_be_visible()
        enable = owner_session.get_by_role("button", name="Edit")
>       expect(enable).to_be_visible()
E       AssertionError: Locator expected to be visible
E       Actual value: None 
E       Call log:
E       LocatorAssertions.to_be_visible with timeout 5000ms
E       waiting for get_by_role("button", name="Edit")
```

**Comment by yohanboniface** [comment]
> Playwright is failing with stuff like:

May be that the HTML is not the map page one, but a 404 or anything like that ?

File: .github/workflows/test-docs.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      if: matrix.dependencies == 'minimal'
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install libgdal-dev
+        python -m pip install --upgrade pip
+        python -m pip install -e ".[dev,test]"
+        playwright install --with-deps
+    - name: run tests
+      run: pytest .
```
Comment (yohanboniface) [inline .github/workflows/test-docs.yml:50]: I'd use `-vv` to have greater details and the test name in the list ? Maybe even use `make test` (which is `-vx`, but why not) ?

Comment (almet) [reply .github/workflows/test-docs.yml]: Failing fast would help, I think, and verbose means more info. always good. Let's do this :-)

File: pyproject.toml [hunk 2]
```diff
@@ -47,17 +45,20 @@ dependencies = [
...
   "djlint==1.31.0",
   "mkdocs==1.5.2",
+  "vermin==1.5.2",
+  "pymdown-extensions==10.4",
+  "isort==5.12",
 ]
 test = [
   "factory-boy==3.2.1",
-  "playwright==1.38.0",
+  "playwright>=1.39,<2",
```
Comment (almet) [inline pyproject.toml:57]: This is needed to be able to support python 3.12.

File: pyproject.toml [hunk 3]
```diff
@@ -23,16 +23,14 @@ classifiers = [
...
-  "Programming Language :: Python :: 3.7",
   "Programming Language :: Python :: 3.8",
   "Programming Language :: Python :: 3.9",
   "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
 ]
 dependencies = [
-  "Django>=4.1",
+  "Django>=4.2,<5",
```
Comment (almet) [inline pyproject.toml:33]: The test suite fails otherwise with Django 4.1 (see #1426)

File: pyproject.toml [hunk 4]
```diff
@@ -47,17 +45,20 @@ dependencies = [
...
 test = [
   "factory-boy==3.2.1",
-  "playwright==1.38.0",
+  "playwright>=1.39,<2",
   "pytest==6.2.5",
   "pytest-django==4.5.2",
-  "pytest-playwright==0.4.2",
-
+  "pytest-playwright>=0.4.3,<1",
+  "pytest-xdist>=3.5.0,<4",
```
Comment (almet) [inline pyproject.toml:61]: This is useful to speedup the tests.

File: umap/tests/settings.py [hunk 5]
```diff
@@ -5,15 +5,17 @@
 SECRET_KEY = "justfortests"
 COMPRESS_ENABLED = False
 FROM_EMAIL = "test@test.org"
-EMAIL_BACKEND = 'django.core.mail.backends.locmem.EmailBackend'
+EMAIL_BACKEND = "django.core.mail.backends.locmem.EmailBackend"
 
-if "TRAVIS" in os.environ:
+if os.environ.get("GITHUB_ACTIONS", False) == "true":
```
Comment (almet) [inline umap/tests/settings.py:10]: Configuration for Github Actions.

File: umap/tests/test_views.py [hunk 6]
```diff
@@ -1,17 +1,18 @@
 import json
 import socket
-from datetime import date, timedelta
+from datetime import date, datetime, timedelta
```
Comment (almet) [inline umap/tests/test_views.py:3]: This fixes warning about datetime that should be time aware.

File: umap/settings/base.py [hunk 7]
```diff
@@ -244,9 +244,7 @@
 UMAP_MAPS_PER_SEARCH = 25
 UMAP_MAPS_PER_PAGE_OWNER = 10
 UMAP_SEARCH_CONFIGURATION = "simple"
-UMAP_FEEDBACK_LINK = (
-    "https://wiki.openstreetmap.org/wiki/UMap#Feedback_and_help"  # noqa
-)
+UMAP_FEEDBACK_LINK = "https://wiki.openstreetmap.org/wiki/UMap#Feedback_and_help"  # noqa
```
Comment (davidbgk) [inline umap/settings/base.py:247]: Is the `# noqa` still relevant?

Comment (almet) [reply umap/settings/base.py:247]: fixed

**Review by davidbgk — APPROVED** [review]
🎁 🙌

**Pull request merged**